### PR TITLE
Notebook for new blog post on versioning

### PR DESCRIPTION
**Pull request by changhiskhan**
includes some minor refactoring to dataset conversion API

### CODE REVIEW

File: integration/duckdb/src/lance/duckdb/video_functions.cc [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+#include <cstdint>
+#include <duckdb/parser/parsed_data/create_scalar_function_info.hpp>
+#include <iostream>
+#include <memory>
+
+namespace lance::duckdb {
+
+template <typename T>
+void L2Distance(::duckdb::DataChunk &args,
```
Comment (eddyxu) [inline integration/duckdb/src/lance/duckdb/video_functions.cc:25]: what is the difference of this to https://github.com/eto-ai/lance/blob/main/integration/duckdb/src/lance/duckdb/vector_functions.cc#L25

File: integration/duckdb/src/lance/duckdb/video_functions.h [hunk 2]
```diff
@@ -0,0 +1,8 @@
+//
```
[the remark this answers is not in the document]
Comment (changhiskhan) [reply integration/duckdb/src/lance/duckdb/video_functions.h:1]: shoot, that's actually the next blog post. let me remove this

File: integration/duckdb/src/lance/duckdb/video_functions.cc (hunk 1 above)
Comment (changhiskhan) [reply integration/duckdb/src/lance/duckdb/video_functions.cc:25]: accident

**Review by eddyxu — APPROVED** [review]
LGTM

**Pull request merged**